qt-models/diveplannermodel.cpp

```diff
@@ -963,8 +968,9 @@ struct divedatapoint * DivePlannerPointsModel::cloneDiveplan(struct diveplan *pl
 	(*dp) = NULL;
 
 	last_segment = plan_copy->dp;
-	while (last_segment->next->next)
+	while (last_segment && last_segment->next && last_segment->next->next)
```

Comment:
Why is this change needed?

Reply:
Otherwise, there is potential crash if if last_segment->next is NULL upon trying to access NULL->next.